lxd/devlxd_instances.go

```diff
@@ -0,0 +1,21 @@
+package main
+
+import (
+	"github.com/canonical/lxd/lxd/device/filters"
+	"github.com/canonical/lxd/lxd/instance"
+)
+
+type devLXDDeviceAccessValidator func(device map[string]string) bool
+
+// newDevLXDDeviceAccessValidator returns a device validator function that
+// checks if the given device is accessible by the devLXD.
+//
+// For example, disk device is accessible if the appropriate security flag
+// is enabled on the instance and the device represents a custom volume.
+func newDevLXDDeviceAccessValidator(inst instance.Instance) devLXDDeviceAccessValidator {
+	diskDeviceAllowed := hasInstanceSecurityFeatures(inst, devLXDSecurityMgmtVolumesKey)
+
+	return func(device map[string]string) bool {
+		return filters.IsCustomVolumeDisk(device) && diskDeviceAllowed
```

Comment:
Suggest switching round the conditions here as its cheaper to check a boolean first.